### PR TITLE
Make working copy customizable

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/repo.rs [hunk 1]
```diff
@@ -119,47 +119,47 @@ pub enum RepoInitError {
 }
 
 impl ReadonlyRepo {
-    pub fn default_op_store_factory() -> impl FnOnce(&Path) -> Box<dyn OpStore> {
-        |store_path| Box::new(SimpleOpStore::init(store_path))
+    pub fn default_op_store_initializer() -> Box<OpStoreInitializer> {
+        Box::new(|store_path| Box::new(SimpleOpStore::init(store_path)))
```
Comment (yuja) [inline lib/src/repo.rs:123]: nit: doesn't have to be boxed?

`-> &'static OpStoreInitializer` works so long as the closure can be turned into function pointer.

File: lib/src/workspace.rs [hunk 2]
```diff
@@ -141,9 +139,9 @@ impl Workspace {
         user_settings: &UserSettings,
         workspace_root: &Path,
     ) -> Result<(Self, Arc<ReadonlyRepo>), WorkspaceInitError> {
-        Self::init_with_backend(user_settings, workspace_root, |store_path| {
-            Ok(Box::new(LocalBackend::init(store_path)))
-        })
+        let backend_initializer: Box<BackendInitializer> =
+            Box::new(|store_path| Ok(Box::new(LocalBackend::init(store_path))));
+        Self::init_with_backend(user_settings, workspace_root, backend_initializer.as_ref())
```
Comment (yuja) [inline lib/src/workspace.rs:149]: nit: here Box isn't needed

**Review by yuja — APPROVED** [review]
I haven't reviewed thoroughly, but the changes look good to me.

File: lib/src/repo.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/repo.rs]: Ah, I didn't think of making it `&'static`. Done.

File: lib/src/workspace.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/workspace.rs]: Done.

**Pull request merged**